### PR TITLE
Clio 39560 actionbar improvements

### PR DESCRIPTION
**Pull request by JefCurtis**

### CODE REVIEW

File: themis_components/thTableActionBar/thTableActionBar.directive.spec.coffee [hunk 1]
```diff
@@ -115,3 +123,17 @@ describe 'ThemisComponents: Directive: thTableActionBar', ->
 
       element.find("button").triggerHandler "click"
       expect(element.isolateScope().actionBar.triggerApply).toHaveBeenCalled()
+
+  context "when 'availableActions' are provided", ->
+    it "the dropdown select is not rendered", ->
```
Comment (slapheadted) [inline themis_components/thTableActionBar/thTableActionBar.directive.spec.coffee]: Minor point, for improved readability, I'd wrap the following two `it` blocks with a `describe "dropdown select"`

`describe "dropdown select"`
`context "when 'availableActions are provided"...`
`it "is not rendered"...`

File: themis_components/thTableActionBar/thTableActionBar.directive.spec.coffee [hunk 2]
```diff
@@ -115,3 +123,17 @@ describe 'ThemisComponents: Directive: thTableActionBar', ->
...
+      setUp()
+      scope.$apply -> scope.delegate.selectPage()
+
+      expect(element.find("select").length).toBe 1
+
+  context "when no 'availableActions' are provided", ->
+    it "the dropdown select is not rendered", ->
```
Comment (slapheadted) [inline themis_components/thTableActionBar/thTableActionBar.directive.spec.coffee]: Same as [above](https://github.com/clio/lib-themisui/pull/180/commits/74c6325d56c8d19a67d362be76a11c1592ceb047#r74474436).

`it "does not render the dropdown select"

File: themis_components/thTableActionBar/thTableActionBar.template.html [hunk 3]
```diff
@@ -37,22 +37,28 @@
           </a>
         </span>
       </div>
-      <th-select
-        ng-model="actionBar.delegate.results.selectedAction"
-        options="actionBar.delegate.results.availableActions"
-        placeholder="Action"
-        condensed="true"
-        ng-disabled="actionBar.delegate.results.processing"
+      <span ng-if="actionBar.delegate.results.availableActions">
```
Comment (slapheadted) [inline themis_components/thTableActionBar/thTableActionBar.template.html]: Asked @JefCurtis about this. He explained to me about the `replace` on the `thSelect` directive causing the issue whereby we have to wrap the `thSelect` element with a span for the `ng-if`.

`#component-hide-show`

File: themis_components/thTableActionBar/readme.md [hunk 4]
```diff
@@ -17,146 +17,133 @@ a Themis themed styling to the link. Check out example 2 to see this in action.
...
+      * `trackedCollection`: This collection is context aware.
+      * `allSelected`: A `boolean` flag indicating the selected state of the
       entire collection. See below for details.
-
-        - When `allSelected` is `true`: `trackedCollection` will start off empty
+        * When `allSelected` is `true`: `trackedCollection` will start off empty
         and include **all the unselected items** as they are unselected.
-
-        - When `allSelected` is `false`: This collection contains **all
+        * When `allSelected` is `false`: This collection contains **all
```
Comment (slapheadted) [inline themis_components/thTableActionBar/readme.md]: (Although not introduced as part of this PR)

On reading this section for the first time I think changing "This collection" to be `trackedCollection` removes any ambiguity between the two terms.

File: themis_components/thTableActionBar/readme.md [hunk 5]
```diff
@@ -17,146 +17,133 @@ a Themis themed styling to the link. Check out example 2 to see this in action.
...
         selected items** as they are selected.
-
-        - Use the value of `allSelected` to determine which collections you are
-        receiving to act on them correctly. `true` means a **negative** collection
-        and `false` means a **positive** collection.
-
-      - `selectedAction` is a string indicating the chosen action.
-
-    - `triggerReset` is a callback function to let the action bar know that it should
+        * Use the value of `allSelected` to determine which collections you are
```
Comment (slapheadted) [inline themis_components/thTableActionBar/readme.md]: "which collections" - does this mean there are more collections than `trackedCollection`? Or do you mean the different states the `trackedCollection` can be in?

File: themis_components/thTableActionBar/thTableActionBar.directive.spec.coffee (hunk 1 above)
Comment (JefCurtis) [inline themis_components/thTableActionBar/thTableActionBar.directive.spec.coffee]: After reading this it was incorrect anyways. Update with fix.

File: themis_components/thTableActionBar/readme.md (hunk 5 above)
Comment (JefCurtis) [inline themis_components/thTableActionBar/readme.md]: There is only one collection `trackedCollection`. The `allSelected` `boolean` flag is used to decipher the state of that collection as explain in the lines above this one. I've added the TLDR as well for clarity. Does this make sense?

Comment (slapheadted) [inline themis_components/thTableActionBar/readme.md]: Nice one.

File: themis_components/thTableActionBar/thTableActionBar.directive.spec.coffee [hunk 6]
```diff
@@ -125,14 +125,14 @@ describe 'ThemisComponents: Directive: thTableActionBar', ->
       expect(element.isolateScope().actionBar.triggerApply).toHaveBeenCalled()
 
   context "when 'availableActions' are provided", ->
-    it "the dropdown select is not rendered", ->
+    it "renders the dropdown select", ->
```
Comment (slapheadted) [inline themis_components/thTableActionBar/thTableActionBar.directive.spec.coffee]: 👍

File: themis_components/thTableActionBar/thTableActionBar.directive.spec.coffee [hunk 7]
```diff
@@ -125,14 +125,14 @@ describe 'ThemisComponents: Directive: thTableActionBar', ->
...
       setUp()
       scope.$apply -> scope.delegate.selectPage()
 
       expect(element.find("select").length).toBe 1
 
   context "when no 'availableActions' are provided", ->
-    it "the dropdown select is not rendered", ->
+    it "does not render the dropdown select", ->
```
Comment (slapheadted) [inline themis_components/thTableActionBar/thTableActionBar.directive.spec.coffee]: 👍

**Pull request merged**